dspace-server-webapp/src/main/java/org/dspace/app/rest/PrimaryBitstreamController.java

```diff
@@ -0,0 +1,161 @@
+     * </code>
+     *
+     * @param uuid      The UUID of the Bundle on which the primaryBitstream will be set
+     * @param request   The HttpServletRequest
+     * @return          The Bundle on which the primaryBitstream was set
+     */
+    @PreAuthorize("hasPermission(#uuid, 'BUNDLE', 'WRITE')")
+    @RequestMapping(method = RequestMethod.POST, consumes = {"text/uri-list"})
+    public ResponseEntity<RepresentationModel<?>> createPrimaryBitstream(@PathVariable UUID uuid,
+                                                                         HttpServletRequest request) {
+        Context context = ContextUtil.obtainContext(request);
+        BundleRest bundleRest = repository.createPrimaryBitstream(context, uuid,
+                                                                  getBitstreamFromRequest(context, request),
+                                                                  utils.obtainProjection());
+        return ControllerUtils.toResponseEntity(HttpStatus.CREATED, new HttpHeaders(),
+                                                (RepresentationModel<?>) converter.toResource(bundleRest));
+    }
+
+    /**
+     * This method updates the primaryBitstream on the given Bundle.
+     * <br><code>
+     * curl -i -X PUT "http://{dspace.server.url}/api/core/bundles/{bundle-uuid}/primaryBitstream"
+     * -H "Content-type:text/uri-list"
+     * -d "https://{dspace.server.url}/api/core/bitstreams/{bitstream-uuid}"
+     * </code>
+     *
+     * @param uuid      The UUID of the Bundle of which the primaryBitstream will be updated
+     * @param request   The HttpServletRequest
+     * @return          The Bundle of which the primaryBitstream was updated
+     */
+    @PreAuthorize("hasPermission(#uuid, 'BUNDLE', 'WRITE')")
+    @RequestMapping(method = RequestMethod.PUT, consumes = {"text/uri-list"})
+    public BundleResource updatePrimaryBitstream(@PathVariable UUID uuid,
+                                                 HttpServletRequest request) {
+        Context context = ContextUtil.obtainContext(request);
+        BundleRest bundleRest = repository.updatePrimaryBitstream(context, uuid,
+                                                                  getBitstreamFromRequest(context, request),
+                                                                  utils.obtainProjection());
+        return converter.toResource(bundleRest);
+    }
+
+    /**
+     * This method deletes the primaryBitstream on the given Bundle.
+     * <br><code>
+     * curl -i -X DELETE "http://{dspace.server.url}/api/core/bundles/{bundle-uuid}/primaryBitstream"
+     * </code>
+     *
+     * @param uuid      The UUID of the Bundle of which the primaryBitstream will be deleted
+     * @param request   The HttpServletRequest
+     * @return          The Bundle of which the primaryBitstream was deleted
+     */
+    @PreAuthorize("hasPermission(#uuid, 'BUNDLE', 'DELETE')")
```

Comment:
I *think* this check should be Bundle *WRITE*.  You aren't deleting the bundle here or the bitstream. Instead, this delete is just "unlinking" a primary Bitstream.

Reply:
@tdonohue I agree that it should be a bundle WRITE and not a DELETE. Good catch.
It also adheres to the mentions of nothing actually being deleted in the [rest contract](https://github.com/DSpace/RestContract/pull/222/files#diff-7bc6f19f04c792846e627a66159dfe67081c3362060319905f0c2d8ba317b6e3R343)